pytd/writer.py

```diff
@@ -17,13 +17,27 @@
 
 
 def _to_list(ary):
+    is_null_ary = pd.isnull(ary)
+    if ary is None or (not isinstance(is_null_ary, np.ndarray) and is_null_ary):
+        return None
     _ary = np.asarray(ary)
+    if ary is None or pd.isnull(_ary).all():
+        return None
     # Replace numpy.nan to None which will be converted to NULL on TD
     if _ary.dtype.kind == "f":
         _ary = np.where(np.isnan(_ary), None, _ary)
     return _ary.tolist()
 
 
+def _check_column_type(x, t):
```

Comment:
Looks like what the function does is very generic one, so how about naming more appropriately like... `_isinstance_or_null`?

```py
def _isinstance_or_null(x, t):
    return isinstance(x, t) or _isnull(x)

def _isnull(x):
    return x is None or x is np.nan
```

This way, `_isnull` allows `_to_list` to simplify some conditions.

Reply:
nice suggestion! I use this idea d97e189